wire/core/WireMail.php

```diff
@@ -328,11 +328,12 @@ public function param($value) {
 	 * @return this 
 	 *
 	 */
-	public function attachment($value) {
+	public function attachment($value, $filename = '') {
 		if(is_null($value)) {
 			$this->mail['attachments'] = array();
 		} else if(is_file($value)) { 
-			$this->mail['attachments'][] = $value; 
+			$filename = $filename ?: basename($value);
+			$this->mail['attachments'][$filename] = $value; 
```

Comment:
Do you see a use-case for multiple attachments named the same? I'd imagine 1 or 2 files max will be the bulk of mails. I see it rather than a case of being able to switch out a file without clearing the whole array of attachments.